analysis/dm_reg_dataset_milan.py

```diff
@@ -0,0 +1,82 @@
+from pathlib import Path
+
+from ehrql import create_dataset
+from ehrql.codes import codelist_from_csv
+
+from ehrql.tables.tpp import (
+    clinical_events,
+    patients,
+    practice_registrations,
+)
+
+CODELIST_DIR = Path("codelists")
+
+# Cluster name: DM_COD
+# Description: Diabetes mellitus codes
+# Refset ID: ^999004691000230108
+dm_cod = codelist_from_csv(
+    CODELIST_DIR / "nhsd-primary-care-domain-refsets-dm_cod.csv",
+    column="code",
+)
+
+# Cluster name: DMRES_COD
+# Description: Diabetes resolved codes
+# Refset ID: ^999003371000230102
+dm_res_cod = codelist_from_csv(
+    CODELIST_DIR / "nhsd-primary-care-domain-refsets-dmres_cod.csv",
+    column="code",
+)
+
+
+# Helper function for finding the last matching event
+def last_matching_event(events, codelist, where=True):
+    return (
+        events.where(where)
+        .where(events.snomedct_code.is_in(codelist))
+        .sort_by(events.date)
+        .last_for_patient()
+    )
+
+
+index_date = "2024-03-31"
+dataset = create_dataset()
+
+has_registration = practice_registrations.for_patient_on(index_date).exists_for_patient()
+dataset.define_population(has_registration)
+
+# Extract prior events for further use in variable definitions below
+prior_events = clinical_events.where(clinical_events.date.is_on_or_before(index_date))
+
+# Field number: 2
+# REG_DAT: The most recent date that the patient registered for GMS, where
+# this registration occurred on or before the achievement date.
+dataset.reg_dat = practice_registrations.for_patient_on(index_date).start_date
```

Comment:
REG_DAT seems to be an unused variable, is there a reason this was included?

Reply:
correct, thanks for spotting this!